### PR TITLE
fix(`@sap-ux/odata-service-inquirer`): Invalid partial dest service paths are allowed

### PR DESCRIPTION
**Pull request by IainSAP**
Fix for #2616 

- Adds more path validations
- Ensure service validation prompt is not shown until a valid path is input
- Adds more tests

### CODE REVIEW

File: packages/odata-service-inquirer/src/prompts/connectionValidator.ts [hunk 1]
```diff
@@ -590,16 +590,23 @@ export class ConnectionValidator {
...
         this._destination = destination;
         // No need to apply sap-client as this happens automatically (from destination config) when going through the BAS proxy
         const status = await this.checkUrl(new URL(destUrl), undefined, undefined, {
             odataVersion: requiredOdataVersion
         });
-        this._validatedUrl = destUrl;
+
         const validationResult = this.getValidationResultFromStatusCode(status);
 
+        if (this.validity.reachable && (!this.validity.authRequired || this.validity.authenticated)) {
```
Comment (longieirl) [inline packages/odata-service-inquirer/src/prompts/connectionValidator.ts:606]: Does this handle the scearnio where for a SAMLAssertion destination, a HTTP 401 shouldn't prompt the user for their credentials?

Comment (IainSAP) [reply packages/odata-service-inquirer/src/prompts/connectionValidator.ts:606]: Not sure I follow the question. There is no prompting here. That would be a different part of the code . This is to ensure we dont add the url as validated if it is not reachable or needs auth.

Comment (longieirl) [reply packages/odata-service-inquirer/src/prompts/connectionValidator.ts:606]: Thanks @IainSAP I will create a separate issue.

**Review by longieirl — APPROVED** [review]
Changes look good.
Issues addressed.
Tests are good.

**Review by devinea — APPROVED** [review]
Changes look good. 
Changeset ✅

**Review by cianmSAP — APPROVED** [review]
Fix looks good
Tests added
Patch changeset

**Pull request merged**